api/checluster_conversion_to_test.go

```diff
@@ -0,0 +1,412 @@
+	assert.Equal(t, "CheServerRoute.Domain", checlusterv2.Spec.Networking.Domain)
+	assert.Equal(t, "CheHost", checlusterv2.Spec.Networking.Hostname)
+	assert.Equal(t, map[string]string{"a": "b", "c": "d"}, checlusterv2.Spec.Networking.Labels)
+	assert.Equal(t, "CheHostTLSSecret", checlusterv2.Spec.Networking.TlsSecretName)
+}
+
+func TestConvertToIngressOnK8s(t *testing.T) {
+	infrastructure.InitializeForTesting(infrastructure.Kubernetes)
+
+	checlusterv1 := &chev1.CheCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "eclipse-che",
+			Namespace: "eclipse-che",
+		},
+		Spec: chev1.CheClusterSpec{
+			Server: chev1.CheClusterSpecServer{
+				CheHost: "CheHost",
+				CheServerIngress: chev1.IngressCustomSettings{
+					Labels:      "a=b,c=d",
+					Annotations: map[string]string{"a": "b", "c": "d"},
+				},
+			},
+			K8s: chev1.CheClusterSpecK8SOnly{
+				IngressDomain: "k8s.IngressDomain",
+				IngressClass:  "k8s.IngressClass",
+				TlsSecretName: "k8s.TlsSecretName",
+			},
+		},
+	}
+
+	checlusterv2 := &chev2.CheCluster{}
+	err := checlusterv1.ConvertTo(checlusterv2)
+	assert.Nil(t, err)
+
+	assert.Equal(t, map[string]string{"a": "b", "c": "d", "kubernetes.io/ingress.class": "k8s.IngressClass"}, checlusterv2.Spec.Networking.Annotations)
+	assert.Equal(t, "k8s.IngressDomain", checlusterv2.Spec.Networking.Domain)
+	assert.Equal(t, "CheHost", checlusterv2.Spec.Networking.Hostname)
+	assert.Equal(t, map[string]string{"a": "b", "c": "d"}, checlusterv2.Spec.Networking.Labels)
+	assert.Equal(t, "k8s.TlsSecretName", checlusterv2.Spec.Networking.TlsSecretName)
+}
+
+func TestConvertToIngressOnK8sWithCheHostTLSSecret(t *testing.T) {
+	infrastructure.InitializeForTesting(infrastructure.Kubernetes)
+
+	checlusterv1 := &chev1.CheCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "eclipse-che",
+			Namespace: "eclipse-che",
+		},
+		Spec: chev1.CheClusterSpec{
+			Server: chev1.CheClusterSpecServer{
+				CheHost:          "CheHost",
+				CheHostTLSSecret: "CheHostTLSSecret",
+				CheServerIngress: chev1.IngressCustomSettings{
+					Labels:      "a=b,c=d",
+					Annotations: map[string]string{"a": "b", "c": "d"},
+				},
+			},
+			K8s: chev1.CheClusterSpecK8SOnly{
+				IngressDomain: "k8s.IngressDomain",
+				IngressClass:  "k8s.IngressClass",
+				TlsSecretName: "k8s.TlsSecretName",
+			},
+		},
+	}
+
+	checlusterv2 := &chev2.CheCluster{}
+	err := checlusterv1.ConvertTo(checlusterv2)
+	assert.Nil(t, err)
+
+	assert.Equal(t, map[string]string{"a": "b", "c": "d", "kubernetes.io/ingress.class": "k8s.IngressClass"}, checlusterv2.Spec.Networking.Annotations)
+	assert.Equal(t, "k8s.IngressDomain", checlusterv2.Spec.Networking.Domain)
+	assert.Equal(t, "CheHost", checlusterv2.Spec.Networking.Hostname)
+	assert.Equal(t, map[string]string{"a": "b", "c": "d"}, checlusterv2.Spec.Networking.Labels)
+	assert.Equal(t, "CheHostTLSSecret", checlusterv2.Spec.Networking.TlsSecretName)
+}
+
+func TestConvertTo(t *testing.T) {
```

Comment:
It may also be worth including a test cases for
1. Round trip conversion (convert v1 -> v2 -> v1 and assert that the before and after are DeepEqual) -- this is the underlying k8s API requirement for conversion
2. Fuzzed objects with round-trip conversion, to ensure e.g. a field added to v2 isn't dropped if that CR is converted v2 -> v1 -> v2